generic-ec-zkp/src/dlog_eq.rs

```diff
@@ -0,0 +1,277 @@
+//! Knowledge of equality of two discrete logarithms $\Pi^\text{dlog_eq}$
+//!
+//! This is a protocol that lets the prover $\P$ convince the
+//! verifier $\V$ that it knows the discrete logarithms of two values, and that
+//! those logarithms are equal. $\P$ knows the secret data $x$ - the discrete
+//! logarithm. $\P$ and $\V$ share common data:
+//! - $H$ - a point on an elliptic curve
+//! - $X = x \cdot G$
+//! - $Z = x \cdot H$
```

Comment:
It is not necessary that $G$ is (the) generator. In general, common data is:

- Points $H_1, H_2$
- $P_1 = x \cdot H_1$
- $P_2 = x \cdot H_2$

And prover proves that it knows $x_1 = \log_{H_1} P_1$ and $x_2 = \log_{H_2} P_2$ and proves that $x_1 = x_2$

Reply:
Fixed